pkg/scheduler/core/util.go

```diff
@@ -68,20 +69,33 @@ func calAvailableReplicas(clusters []*clusterv1alpha1.Cluster, spec *workv1alpha
 		return availableTargetClusters
 	}
 
-	// Get the minimum value of MaxAvailableReplicas in terms of all estimators.
+	// get estimators and sort it by priority.
 	estimators := estimatorclient.GetReplicaEstimators()
+	sort.Slice(estimators, func(i, j int) bool {
+		return estimators[i].GetPriority() > estimators[j].GetPriority()
+	})
+
+	priorityOfAvailableEstimator := estimatorclient.EstimatorPriority(math.MinInt32)
 	ctx := context.WithValue(context.TODO(), util.ContextKeyObject,
 		fmt.Sprintf("kind=%s, name=%s/%s", spec.Resource.Kind, spec.Resource.Namespace, spec.Resource.Name))
+
 	for _, estimator := range estimators {
+		// if higher-priority estimator has given the result, no longer to call lower-priority estimator.
+		if estimator.GetPriority() < priorityOfAvailableEstimator {
+			break
+		}
 		res, err := estimator.MaxAvailableReplicas(ctx, clusters, spec.ReplicaRequirements)
 		if err != nil {
 			klog.Errorf("Max cluster available replicas error: %v", err)
 			continue
 		}
+		priorityOfAvailableEstimator = estimator.GetPriority()
 		for i := range res {
+			// if an estimator can only give a part of result of member clusters, same priority estimator will be called, but low-priority estimator not.
```

Comment:
I think low-priority estimator may be called. We would rather choose lower precision results than not return results at all. And I noticed that there are some corner cases here.
### Eg:
The priority of estimators(`A, B, C`) are `3, 3, 2`.
* Estimator `A` only gave a part of result of member clusters, `C` should be called when`B` failed or gave the partial result.
* Estimator `A` gave the full result of member clusters, `C` shouldn't be called when `B` failed or gave the partial result.
* Estimator `A` and `B` both gave the partial result, but when they can form a complete result, `C` shouldn't be called.

Reply:
@whitewindmills If you want to cover all corner cases, things can be more complex, like:

The priority of estimators(A, B, C, D) are 3, 3, 2, 2.

* Estimator A and B both gave the partial result and can't form a complete result, but after call C, they form a complete result, however D has the same priority with C, now should D be called?

Reply:
Of course `D` should be called. We cannot guarantee that all corner cases will be covered. That's just what I can think of right now. From another perspective, this PR may not need to be considered so completely, just follow the simple fallback principle.

Reply:
done, thanks